### PR TITLE
fix(v2): Update migrated and recalled data sets in tree

### PR DESCRIPTION
**Pull request by traeok**
## Proposed changes

Fixes the following issues:

- Fixed an issue where a migrated data set is unusable after it is recalled through Zowe Explorer
- Fixed an issue where a recalled PDS is expandable after it is migrated through Zowe Explorer
- Fixed an issue where data set nodes did not update if migrated or recalled outside of Zowe Explorer

## Release Notes

Milestone: 2.18.1

Changelog:

- Fixed an issue where a migrated data set is unusable after it is recalled through Zowe Explorer. [#3294](https://github.com/zowe/zowe-explorer-vscode/issues/3294)
- Fixed an issue where a recalled PDS is expandable after it is migrated through Zowe Explorer. [#3294](https://github.com/zowe/zowe-explorer-vscode/issues/3294)
- Fixed an issue where data set nodes did not update if migrated or recalled outside of Zowe Explorer. [#3294](https://github.com/zowe/zowe-explorer-vscode/issues/3294)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (non-breaking change which adds or improves functionality)
- [ ] Breaking change (a change that would cause existing functionality to not work as expected)
- [ ] Documentation (Markdown, README updates)
- [ ] Other (please specify above in "Proposed changes" section)

## Checklist

**General**

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/zowe-explorer-vscode/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] All PR dependencies have been merged and published (if applicable)
- [ ] A GIF or screenshot is included in the PR for visual changes
- [x] The pre-publish command has been executed:
  - **v2 and below:** `yarn workspace vscode-extension-for-zowe vscode:prepublish`
  - **v3:** `pnpm --filter vscode-extension-for-zowe vscode:prepublish`

**Code coverage**

- [x] There is coverage for the code that I have added
- [x] I have added new test cases and they are passing
- [x] I have manually tested the changes

**Deployment**

- [ ] I have added developer documentation (if applicable)
- [ ] Documentation should be added to Zowe Docs
  - If you're an outside contributor, please post in the [#zowe-doc Slack channel](https://openmainframeproject.slack.com/archives/CC961JYMQ) to coordinate documentation.
  - Otherwise, please check with the rest of the squad about any needed documentation before merging.
- [x] These changes may need ported to the appropriate branches (list here): `main` (WIP)

### CODE REVIEW

**Comment by traeok** [comment]
"1 New Issue" is an existing problem with `ZoweDatasetNode` complexity. This should be resolved separately.

**Review by zFernand0 — DISMISSED** [review]
Had a chance to test this and it seems to work well.

All things considered, LGTM! 😋 

---
From an internal conversation 😋 

_The migration request is sent and the icon changes automatically (which could technically be wrong 😅)
If I do the same dataset search, I still see it there (not-migrated)
But that's just because of how migration/archival schedulers may be configured._

**Comment by traeok** [comment]
> Had a chance to test this and it seems to work well.
> 
> All things considered, LGTM! 😋
> 
> From an internal conversation 😋
> 
> _The migration request is sent and the icon changes automatically (which could technically be wrong 😅) If I do the same dataset search, I still see it there (not-migrated) But that's just because of how migration/archival schedulers may be configured._

Thanks for testing Fernando and sorry for quickly dismissing your review 😅  I've addressed the behavior you mentioned where the icon changes right away. Now, we won't change the icon and we'll wait for it to be reflected when listing data sets.

**Review by zFernand0 — APPROVED** [review]
Should we update the version of imperative/cli to get the updated cross-spawn 😋 

LGTM! 😋

**Comment by traeok** [comment]
> Should we update the version of imperative/cli to get the updated cross-spawn 😋
> 
> LGTM! 😋

Once [zowe-cli#2374](https://github.com/zowe/zowe-cli/pull/2374) is merged, happy to bump the SDK/imperative versions in this PR 😋

**Review by anaxceron — DISMISSED** [review]

File: packages/zowe-explorer/src/dataset/ZoweDatasetNode.ts [hunk 1]
```diff
@@ -131,6 +131,46 @@ export class ZoweDatasetNode extends ZoweTreeNode implements IZoweDatasetTreeNod
         }
     }
 
+    /**
+     * Updates an existing data set node that was recalled so it can be interacted with.
+     * @param isPds Whether the data set is a PDS
+     */
+    private datasetRecalled(isPds: boolean): void {
+        // Change context value to match dsorg, update collapsible state
+        this.contextValue = isPds ? globals.DS_PDS_CONTEXT : globals.DS_DS_CONTEXT;
```
Comment (awharn) [inline packages/zowe-explorer/src/dataset/ZoweDatasetNode.ts:140]: What if it's favorited?

Comment (traeok) [reply packages/zowe-explorer/src/dataset/ZoweDatasetNode.ts:141]: Good call - it looks like in the logic before this PR, we assumed it wasn't a favorite and set the context value based on collapsible state. I addressed this in 1ecdb88.

**Review by zFernand0 — DISMISSED** [review]
LGTM! 😋

**Comment by zFernand0** [comment]
> > Should we update the version of imperative/cli to get the updated cross-spawn 😋
> > LGTM! 😋
> 
> Once [zowe-cli#2374](https://github.com/zowe/zowe-cli/pull/2374) is merged, I'll bump the SDK/imperative versions in this PR 😁

probably have to wait for the new cli version to be on public npm (tomorrow 🙏 )

**Review by awharn — APPROVED** [review]

**Review by zFernand0 — DISMISSED** [review]
LGTM! 😋 
will re-approve once the CLI package is updated 🙏

**Review by JillieBeanSim — APPROVED** [review]
LGTM! adding approval in case we end up wanting to add cli dependency update separately 😄 
Thanks for the fix @traeok

**Review by zFernand0 — APPROVED** [review]

**Pull request merged**